From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dynamic Devices <support@dynamicdevices.co.uk>
Date: Sun, 1 Jun 2026 00:00:00 +0000
Subject: [PATCH] imx9/scmi: export check_secondary_cnt_set for SPL

image-container.c references check_secondary_cnt_set for all i.MX9 builds,
but u-boot-fio imx-2024.04 only implements it for i.MX8 (static in misc.c).
Factory u-boot-fio SPL links scmi/soc.c (CONFIG_SCMI_FIRMWARE) and fails to
link without this symbol. Provide a primary-set stub in scmi/soc.c.
---
 arch/arm/mach-imx/imx9/scmi/soc.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm/mach-imx/imx9/scmi/soc.c b/arch/arm/mach-imx/imx9/scmi/soc.c
index 1111111111..2222222222 100644
--- a/arch/arm/mach-imx/imx9/scmi/soc.c
+++ b/arch/arm/mach-imx/imx9/scmi/soc.c
@@ -44,6 +44,7 @@
 #endif
 #include <spl.h>
 #include <mmc.h>
+#include <imx_container.h>
 #include <kaslr.h>
 
 
@@ -51,6 +52,14 @@ DECLARE_GLOBAL_DATA_PTR;
 
 rom_passover_t rom_passover_data = {0};
 
+bool check_secondary_cnt_set(unsigned long *set_off)
+{
+	if (set_off)
+		*set_off = 0;
+
+	return false;
+}
+
 uint32_t scmi_get_rom_data(rom_passover_t *rom_data)
 {
 	/* Read ROM passover data */
